app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+                "url": "/services/data/v20.0"
+            },
+            {
+                "version": "21.0",
+                "label": "Spring '11",
+                "url": "/services/data/v21.0"
+            },
+            {
+                "version": "26.0",
+                "label": "Winter '13",
+                "url": "/services/data/v26.0"
+            }
+        ]
+        """
+        # instance_url identifies the Salesforce instance to which API calls should be sent.
+        if not self._instance_url:
+            LOGGER.error('No instance url found for %s %s', self.service, self._type)
+            return
+
+        url = '{}/services/data/'.format(self._instance_url)
+        success, response = self._make_get_request(url, self._auth_headers)
+
+        if not success:
+            LOGGER.error('Failed to fetch lastest api version')
+            return
+
+        if response:
+            versions = [float(version.get('version', 0)) for version in response]
+            if versions:
+                self._latest_api_version = str(sorted(versions)[-1])
+                LOGGER.debug('Successfully obtain latest API version %s', self._latest_api_version)
+
+    def _list_log_files(self):
+        """Fetch a list of available log files by event types.
+
+        An event generates log data in real time. However, log files are generated
+        the day after an event takes place, during nonpeak hours. Therefore, log
+        file data is unavailable for at least one day after an event.
+
+        Returns:
+            list: a list of log file location or empty list.
+
+        An example of log files response json body:
+        {
+            "totalSize": 2,
+            "done": True,
+            "records": [
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ"
+                    },
+                    "Id": "0ATD000000001bROAQ",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ/LogFile",
+                    "LogDate": "2014-03-14T00:00:00.000+0000",
+                    "LogFileLength": 2692.0
+                },
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD000000001SdOAI"
+                    },
+                    "Id": "0ATD000000001SdOAI",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001SdOAI/LogFile",
+                    "LogDate": "2014-03-13T00:00:00.000+0000",
+                    "LogFileLength": 1345.0
+                }
+            ]
+        }
+        """
+        log_files = []
+        for event_type in self._SALESFORCE_EVENT_TYPES:
```

Comment:
+++ beautiful!!

Reply:
if you want to speak more about subclassing your new app to handle different log types let me know!!